### PR TITLE
Put an upper bound on bleach and comment each and every dependency

### LINKED ISSUE #1252: Error Due To Bleach Updating
### Discussed in https://github.com/django-wiki/django-wiki/discussions/1251

<div type='discussions-op-text'>

<sup>Originally posted by **Seanld** January 24, 2023</sup>
Mozilla's `bleach` library updated today, and now django-wiki is giving me this error when I try to start my project:

```
everyname-integration-1  | Traceback (most recent call last):
everyname-integration-1  |   File "/app/manage.py", line 15, in <module>
everyname-integration-1  |     execute_from_command_line(sys.argv)
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 425, in execute_from_command_line
everyname-integration-1  |     utility.execute()
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 401, in execute
everyname-integration-1  |     django.setup()
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
everyname-integration-1  |     apps.populate(settings.INSTALLED_APPS)
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/django/apps/registry.py", line 114, in populate
everyname-integration-1  |     app_config.import_models()
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/django/apps/config.py", line 300, in import_models
everyname-integration-1  |     self.models_module = import_module(models_module_name)
everyname-integration-1  |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
everyname-integration-1  |     return _bootstrap._gcd_import(name[level:], package, level)
everyname-integration-1  |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
everyname-integration-1  |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
everyname-integration-1  |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
everyname-integration-1  |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
everyname-integration-1  |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
everyname-integration-1  |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/wiki/models/__init__.py", line 6, in <module>
everyname-integration-1  |     from .article import *  # noqa
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/wiki/models/article.py", line 17, in <module>
everyname-integration-1  |     from wiki.conf import settings
everyname-integration-1  |   File "/usr/local/lib/python3.10/site-packages/wiki/conf/settings.py", line 60, in <module>
everyname-integration-1  |     bleach.ALLOWED_TAGS
everyname-integration-1  | TypeError: unsupported operand type(s) for +: 'frozenset' and 'list'
```</div>

### PR DESCRIPTION
**Pull request by benjaoming**
Maybe it's useful to start commenting on why a dependency is there and what expectations we have for the versions that are pinned? What do you think @oscarmcm ? :)

Fixes #1252

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,13 +37,26 @@ classifiers = [
     "Topic :: Software Development :: Libraries :: Application Frameworks",
 ]
 dependencies = [
+    # Django version support is added one minor release at a time
     "Django>=2.1,<4.2",
-    "bleach[css]>=5",
+    # bleach has been pretty stable, hence the loose pin
+    "bleach[css]>=5,<6",
+    # Pillow is used very little and has never broken
     "Pillow",
```
Comment (oscarmcm) [inline pyproject.toml:45]: Shouldn't we also pin the Pillow version?

**Review by oscarmcm — APPROVED** [review]
LGTM 🥇 

And thank you for the comments in the dependencies 💯!

**Comment by oscarmcm** [comment]
@benjaoming Quick question, do we need to draft a new release after this? I believe this is a breaking change and prevents the app to be used.


Will wait for your answer in order to merge it.

**Comment by benjaoming** [comment]
Yes, this should be a new release. But IIRC, we already made changes in `main` that should bump the minor version :)

File: pyproject.toml [hunk 2]
```diff
@@ -37,13 +37,26 @@ classifiers = [
...
+    # control breakage and pinning ourselves
     "django-nyt>=1.2.2",
+    # django-mptt has had several breaking changes
     "django-mptt>=0.13,<0.15",
+    # django-sekizai is basically only releasing every time
+    # it needs to restore support for a new Django/Python version
     "django-sekizai>=0.10",
+    # sorl-thumbnail is maintained by jazzband so it might be
+    # very stable but it might also suddenly invite breaking changes
     "sorl-thumbnail>=12.8,<13",
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply pyproject.toml:56]: I don't know TBH.

sort-thumbnail has been quite stable. Not a lot of bugs, but maybe it's good to switch to a project that is actively developed.

File: pyproject.toml (hunk 1 above)
Comment (benjaoming) [reply pyproject.toml:45]: Could be good, I don't have strong opinions on this one... I don't even know what the lower bound should be.

Comment (oscarmcm) [reply pyproject.toml:45]: I'll leave this one as it is, bcz most projects already have/use a pinned Pillow version and Django-Wiki will use that version (if its in the right order).

**Comment by oscarmcm** [comment]
Yeah, I believe this should be the workflow we can follow now:

- `hatch version` - Should match the current release version (`0.9`)
- `hatch build`
- `hatch version minor` - If I'm not wrong should be `0.9.1`
- Create a new release tag
- Push the new release tag
- `hatch publish`
- Add the artifacts to the release tag

**Comment by benjaoming** [comment]
> hatch version minor - If I'm not wrong should be 0.9.1

Ah no, that would be a patch version. We should bump to 0.10. This doesn't mean that we'd want to support a 0.9.x series with patch updates, but more to cater for people who expect 0.9.x to not break... because we might break stuff with the series of changes going in here: https://github.com/django-wiki/django-wiki/pulls?q=is%3Apr+is%3Aclosed

I did also kind of skip writing change logs :innocent:  https://github.com/django-wiki/django-wiki/blob/main/docs/release_notes.rst

**Comment by oscarmcm** [comment]
Ah oks, that sounds great! can't wait for this release!!

**Comment by statsconchris** [comment]
Using `pip install --pre wiki` doesn't take this update into account. I keep getting the error. what options do I have

**Comment by oscarmcm** [comment]
Hi @statsconchris I've replied in the discussion https://github.com/django-wiki/django-wiki/discussions/1257#discussioncomment-4773040 maybe that option works for now until we release a new version.

**Pull request merged**